### PR TITLE
allow non-factor statnames in utils::transformstats

### PR DESCRIPTION
**Pull request by infotroph**
## Description

`PEcAn.utils::transformstats` currently assumes the `statname` column if its input is a factor. This patch allows it to be character instead, if desired. Output keeps the same class it had on input.

Edit: while I was here, also cleaned up code formatting and fixed logic in the untransformed stats warning.

## Motivation and Context

Noticed now because the development version of R [no longer coerces character vectors in `data.frame` to factor](https://developer.r-project.org/Blog/public/2020/02/16/stringsasfactors/), so our unit test started failing. Good catch, unit test!

## Review Time Estimate

- [ ] Immediately
- [ ] Within one week
- [ ] When possible
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue) 
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the CHANGELOG.md.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by infotroph** [comment]
@mdietze @dlebauer I think this is ready to merge -- previous Travis failures are hopefully now resolved.

File: base/utils/R/transformstats.R [hunk 1]
```diff
@@ -1,46 +1,52 @@
...
 ##' Transform misc. statistics to SE
 ##'
-##' Automates transformations of SD, MSE, LSD, 95%CI, HSD, and MSD to conservative estimates of SE.
-##' @name transformstats
-##' @title Transform Stats 
-##' @param data data frame with mean, statistic, n, and statistic name: \code{example data <- data.frame(Y=rep(1,5), stat=rep(1,5), n=rep(4,5), statname=c('SD', 'MSE', 'LSD', 'HSD', 'MSD'))}
-##' @return dataframe with statistics transformed to SE
+##' Automates transformations of SD, MSE, LSD, 95%CI, HSD, and MSD
+##'   to conservative estimates of SE.
+##'
```
Comment (dlebauer) [inline base/utils/R/transformstats.R]:
```suggestion
##' Method details and assumptions described in
##' LeBauer 2020 Transforming ANOVA and Regression statistics for Meta-analysis. Authorea. DOI: https://doi.org/10.22541/au.158359749.96662550
```

File: base/utils/R/transformstats.R [hunk 2]
```diff
@@ -1,46 +1,53 @@
...
 ##'
-##' Automates transformations of SD, MSE, LSD, 95%CI, HSD, and MSD to conservative estimates of SE.
-##' @name transformstats
-##' @title Transform Stats 
-##' @param data data frame with mean, statistic, n, and statistic name: \code{example data <- data.frame(Y=rep(1,5), stat=rep(1,5), n=rep(4,5), statname=c('SD', 'MSE', 'LSD', 'HSD', 'MSD'))}
-##' @return dataframe with statistics transformed to SE
+##' Automates transformations of SD, MSE, LSD, 95%CI, HSD, and MSD
+##'   to conservative estimates of SE.
+##' Method details and assumptions described in
+##' LeBauer 2020 Transforming ANOVA and Regression statistics for Meta-analysis. Authorea. DOI: https://doi.org/10.22541/au.158359749.96662550
```
Comment (dlebauer) [inline base/utils/R/transformstats.R]:
```suggestion
##' LeBauer 2020 Transforming ANOVA and Regression statistics for Meta-analysis. 
##' Authorea. DOI: https://doi.org/10.22541/au.158359749.96662550
```

**Comment by dlebauer** [comment]
I still don't understand travis well enough to know what to do with the travis failure  - the log says "build exited with status 1" indicating success, but then travis says the build fails https://travis-ci.org/github/PecanProject/pecan/jobs/664465417?utm_medium=notification&utm_source=github_status

**Comment by infotroph** [comment]
@dlebauer the statuses are from bash, so anything not 0 is a failure. 

Looks like this is yet another unrelated dependency failure, this time in the DB tests: apparently  `PEcAn.DB::build_insert_query` is now returning doubled quotes because of something that changed between `glue` 1.3.1 and 1.3.2

**Comment by ashiklom** [comment]
As I mentioned on Slack, I think the easiest fix here is to replace the `glue_sql` statement in `base/db/R/insert_table.R::build_insert_query` with the following:

```r
insert_string <- paste(insert_list, collapse = ",")
result <- glue::glue_sql(
  "INSERT INTO {`table`} ({`colnames(values)`*}) VALUES ",
  .con = .con
)
DBI::SQL(paste0(result, "(", insert_string, ")"))
```

It feel like a less-robust solution because it replaces supposedly robust backend-dependent SQL quoting with hard-coded quoting...but I _think_ it should work with the SQLite backend in the test. 

Also as I mentioned on Slack, the _ideal_ solution here is to simplify the code in that script to just use prepared statements for everything, which should be faster, safer, and easier to implement.

**Pull request merged**